src/strategies/optimism/velodrome/strategy-velo-base.sol

```diff
@@ -0,0 +1,165 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.6;
+
+import "../strategy-base-v2.sol";
+import "../../../optimism/interfaces/solidly/IRouter.sol";
+import "../../../optimism/interfaces/solidly/IGauge.sol";
+
+abstract contract StrategyVeloBase is StrategyBase {
+    // Addresses
+    address public constant solidRouter = 0x9c12939390052919aF3155f41Bf4160Fd3666A6f;
+    address public constant velo = 0x3c8B650257cFb5f272f799F5e2b4e65093a11a05;
+
+    address public token0;
+    address public token1;
+
+    address public gauge;
+
+    bool public isStablePool;
+
+    mapping(address => ISolidlyRouter.route[]) public nativeToTokenRoutes;
+    mapping(address => ISolidlyRouter.route[]) public toNativeRoutes;
+
+    constructor(
+        address _lp,
+        address _gauge,
+        address _governance,
+        address _strategist,
+        address _controller,
+        address _timelock
+    ) StrategyBase(_lp, _governance, _strategist, _controller, _timelock) {
+        ISolidlyRouter.route[] memory _veloRoute = new ISolidlyRouter.route[](2);
+        _veloRoute[0] = ISolidlyRouter.route(velo, 0x7F5c764cBc14f9669B88837ca1490cCa17c31607, false);
+        _veloRoute[1] = ISolidlyRouter.route(0x7F5c764cBc14f9669B88837ca1490cCa17c31607, weth, false);
+        _addToNativeRoute(_veloRoute);
+
+        gauge = _gauge;
+        token0 = IUniswapV2Pair(_lp).token0();
+        token1 = IUniswapV2Pair(_lp).token1();
+
+        IERC20(native).approve(solidRouter, type(uint256).max);
+        IERC20(token0).approve(solidRouter, type(uint256).max);
+        IERC20(token1).approve(solidRouter, type(uint256).max);
+        IERC20(velo).approve(solidRouter, type(uint256).max);
+        IERC20(want).approve(gauge, type(uint256).max);
+    }
+
+    function balanceOfPool() public view override returns (uint256) {
+        return ISolidlyGauge(gauge).balanceOf(address(this));
+    }
+
+    function getHarvestable() external view override returns (address[] memory, uint256[] memory) {
+        uint256[] memory pendingRewards = new uint256[](1);
```

Comment:
Would it make sense to make the `getHarvestable` dynamic by setting the length of these arrays to be the length of `activeRewardsTokens`? Just anticipating the upcoming $OP rewards, but not sure if enough is known about their distribution by Velo to anticipate it in this function (might just have to ultimately deal with it in pf-core 😅)